dl/src/main/scala/com/intel/analytics/bigdl/utils/LoggerAppender.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to Intel Corporation under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * Intel Corporation licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.utils
+
+import org.apache.log4j._
+
+import scala.collection.JavaConverters._
+
+/**
+ * logger appender, which will output the log to files
+ */
+object LoggerAppender {
+
+  val pattern = "%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n"
+
+  /**
+   * redirect log to `filePath`, whose default value is /tmp/bigdl.log
+   *
+   * @param filePath log file path, default value is /tmp/bigdl.log
+   * @return a new file appender
+   */
+  def fileAppender(filePath: String = "/tmp/bigdl.log"): FileAppender = {
+    val fileAppender = new FileAppender
+    fileAppender.setName("FileLogger")
+    fileAppender.setFile(filePath)
+    fileAppender.setLayout(new PatternLayout(pattern))
+    fileAppender.setThreshold(Level.INFO)
+    fileAppender.setAppend(true)
+    fileAppender.activateOptions()
+
+    fileAppender
+  }
+
+  /**
+   * redirect log to console or stdout
+   *
+   * @return a new console appender
+   */
+  def consoleAppender(): ConsoleAppender = {
+    val console = new ConsoleAppender
+    console.setLayout(new PatternLayout(pattern))
+    console.setThreshold(Level.INFO)
+    console.activateOptions()
+
+    console
+  }
+
+  /**
+   * find the logger of `clz` and add a new appender to it.
+   *
+   * @param clz class which user defined
+   * @param appender appender, eg. return of `fileAppender` or `consoleAppender`
+   */
+  def redirectAppender(clz: String, appender: Appender): Unit = {
+    Logger.getLogger(clz).addAppender(appender)
+  }
+
+  /**
+   * make the level of console appender in root logger to ERROR,
+   * which will interfere with all other loggers.
+   */
+  private def resetLevelToError(): Unit = {
+    val appenders = Logger.getRootLogger.getAllAppenders.asScala
+
+    for (appender <- appenders) {
+      appender match {
+        case a: ConsoleAppender => a.setThreshold(Level.ERROR)
+        case _ => // ignore other appender except ConsoleAppender
+      }
+    }
+  }
+
+  /**
+   * redirect all logs of Spark to fileAppender (/tmp/bigdl.log).
+   * and add an console appender for `com.intel.analytics.bigdl.optim`
+   */
+  def redirectAllAppenders(): Unit = {
+    resetLevelToError()
```

Comment:
1. We can't just set the level of `org`, `breeze` to error because we can't get the `Appender` of these loggers.
2. It will  output all message to console if we do not set the level of root logger to error.
3. I think error message is very important so I don't set the level to OFF.
4. We can set the correct level (INFO) of BigDL after this redirection.